data-plane/receiver/src/test/java/dev/knative/eventing/kafka/broker/receiver/impl/IngressProducerReconcilableStoreTest.java

```diff
@@ -321,4 +323,130 @@ private void testStore(
+      .build();
+
+    final var resource2 = DataPlaneContract.Resource.newBuilder()
+      .setUid("2")
+      .addTopics("topic2")
+      .setBootstrapServers("kafka-2:9092")
+      .setIngress(DataPlaneContract.Ingress.newBuilder().setHost("http://host2").setPath("/")
+        .setContentMode(DataPlaneContract.ContentMode.STRUCTURED).build())
+      .build();
+
+    final var resource3 = DataPlaneContract.Resource.newBuilder()
+      .setUid("3")
+      .addTopics("topic3")
+      .setBootstrapServers("kafka-3:9092")
+      .setIngress(DataPlaneContract.Ingress.newBuilder().setHost("http://host3").setPath("/hello3")
+        .setContentMode(DataPlaneContract.ContentMode.STRUCTURED).build())
+      .build();
+
+    // additional resource to make sure path "/" don't mean anything
+    final var resource4 = DataPlaneContract.Resource.newBuilder()
+      .setUid("4")
+      .addTopics("topic4")
+      .setBootstrapServers("kafka-4:9092")
+      .setIngress(DataPlaneContract.Ingress.newBuilder().setHost("http://host4").setPath("/")
+        .setContentMode(DataPlaneContract.ContentMode.STRUCTURED).build())
+      .build();
+
+    KafkaProducer<String, CloudEvent> producer1 = mockProducer();
+    KafkaProducer<String, CloudEvent> producer2 = mockProducer();
+    KafkaProducer<String, CloudEvent> producer3 = mockProducer();
+    KafkaProducer<String, CloudEvent> producer4 = mockProducer();
+
+    final var store = new IngressProducerReconcilableStore(
+      AuthProvider.noAuth(),
+      new Properties(),
+      properties -> {
+        if ("kafka-1:9092".equals(properties.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG))) {
+          return producer1;
+        } else if ("kafka-2:9092".equals(properties.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG))) {
+          return producer2;
+        } else if ("kafka-3:9092".equals(properties.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG))) {
+          return producer3;
+        } else if ("kafka-4:9092".equals(properties.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG))) {
+          return producer4;
+        }
+        throw new IllegalStateException("Can't determine what producer to return");
+      }
+    );
+
+    final var reconciler = ResourcesReconciler
+      .builder()
+      .watchIngress(store)
+      .build();
+
+    vertx.runOnContext(v -> {
+      Future.succeededFuture()
+        .compose(v1 -> reconciler.reconcile(List.of(resource1, resource2, resource3, resource4)))
+        .onSuccess(event -> {
+          // match by path
+          assertThat(store.resolve("", "/hello1").getKafkaProducer()).isSameAs(producer1);
+          assertThat(store.resolve("", "/hello3").getKafkaProducer()).isSameAs(producer3);
+
+          // match by host
+          assertThat(store.resolve("http://host2", "/").getKafkaProducer()).isSameAs(producer2);
+          assertThat(store.resolve("http://host4", "/").getKafkaProducer()).isSameAs(producer4);
```

Comment:
I think the `/` is optional, so testing for the empty path too would increase coverage or reveal a bug (but we don't write buggy code, right?)
```suggestion
          assertThat(store.resolve("http://host2", "/").getKafkaProducer()).isSameAs(producer2);
          assertThat(store.resolve("http://host4", "/").getKafkaProducer()).isSameAs(producer4); 
          assertThat(store.resolve("http://host2", "").getKafkaProducer()).isSameAs(producer2);
          assertThat(store.resolve("http://host4", "").getKafkaProducer()).isSameAs(producer4); 
```

Reply:
LOL good point :D

Reply:
And I never write buggy code!